packages/eslint-config-typescript/index.js

```diff
@@ -42,5 +42,7 @@ module.exports = {
     // semi interference fix.
     semi: 'off',
     '@typescript-eslint/semi': 'warn',
+    // no discouraged words
+    '@spotify/best-practices/no-discouraged-words': 'error',
```

Comment:
the base config itself is `@spotify/eslint-config`, so we should enable the rule in that project, not `eslint-config-typescript.`

I also think a good way to begin rolling this out would be make the default `warn`;  we can easily make the default `error` in 9.0.0 that way.